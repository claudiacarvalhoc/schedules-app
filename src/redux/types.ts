export const FETCH_SCHEDULE_SUCCESS = 'FETCH_SCHEDULE_SUCCESS';
export const FETCH_SCHEDULE_FAILURE = 'FETCH_SCHEDULE_FAILURE';

export const FETCH_SCHEDULE_LOGS_SUCCESS = 'FETCH_SCHEDULE_LOGS_SUCCESS';
export const FETCH_SCHEDULE_LOGS_FAILURE = 'FETCH_SCHEDULE_LOGS_FAILURE';

export const REQUEST_RETIRE_SCHEDULE = 'REQUEST_RETIRE_SCHEDULE';
export const REQUEST_UNRETIRE_SCHEDULE = 'REQUEST_UNRETIRE_SCHEDULE';
export const UPDATE_SELECTED_SCHEDULE = 'UPDATE_SELECTED_SCHEDULE';