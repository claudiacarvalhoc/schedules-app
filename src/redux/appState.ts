import { DefaultRootState } from "react-redux";

/**
 * The App Status
 */
export interface AppState extends DefaultRootState {
    status: AppStatusSectionsState;
    selectedScheduleId?: number;
    schedules: ScheduleState[];
    scheduleLogs: ScheduleLogState[];
    texts: {
        header: HeaderTextStatus;
        schedules: ScheduleTextState;
    }
}

/**
 * The status to fetch data per section
 */
export interface AppStatusSectionsState {
    schedules: StatusState;
    scheduleLogs: StatusState;
}

/**
 * The Boot Status is defined for options: Loading / Success / Failure.
 */
export enum StatusState {
    Loading,
    Success,
    Failure
}

/**
 * The Schedule interface.
 */
export interface ScheduleState {
    id: number;
    avatar: AvatarState;
    name: string;
    description: string;
    isRetired: boolean;
    tasksCount: number;
    startPoint: Date;
    endPoint: Date;
    dayOfWeek: number;
    dayOfMonth: number;
    startDate: Date;
    endDate: Date;
    schedules?: IntervalTypeState;
    timePeriod?: number;
}

/**
 * The Avatar Identification
 */
export interface AvatarState {
    letter: string;
    color: string;
}

/**
 * The Schedule Interval Types
 */
export enum IntervalTypeState {
    Once,
    Year,
    Week,
    Hour,
    Minute,
    Second
}

/**
 * The Schedule Logs interface.
 */
export interface ScheduleLogState {
    id: number;
    startTime: Date;
    endTime: Date;
    status: LogStatusState;
    serverName: string;
    scheduleId: number;
}

/**
 * The Schedule Logs Status
 */
export enum LogStatusState {
    Terminated,
    Pending,
    Completed,
    Exception,
    Running
}

/**
 * Texts used on Header
 * Note: This allow us to add localisation feature easily
 */
export interface HeaderTextStatus {
    titleText: string;
}

/**
 * Texts used on Schedule
 * Note: This allow us to add localisation feature easily
 */
export interface ScheduleTextState {
    buttonRetireText: string;
    buttonUnretireText: string;
    errorMessageText: string;
    buttonRetryText: string;
    emptyMessageText: string;
}
