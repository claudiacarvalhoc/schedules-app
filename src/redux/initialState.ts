import { AppState, AvatarState, StatusState } from "./appState";

/**
 * This is the initial state of redux.
 */
export const appInitialState: AppState = {
    status: {
        schedules: StatusState.Loading,
        scheduleLogs: StatusState.Loading,
    },
    selectedScheduleId: undefined,
    schedules: [],
    scheduleLogs: [],
    texts: {
        header: {
            titleText: 'Schedules',
        },
        schedules: {
            buttonRetireText: 'Retire',
            buttonUnretireText: 'Unretire',
            errorMessageText: 'Something went wrong ...',
            buttonRetryText: 'Retry',
            emptyMessageText: 'No schedules available.',
        },
        scheduleLogs: {
          errorMessageText: 'Something went wrong ...',
          buttonRetryText: 'Retry',
          emptyMessageText: 'No schedules logs available.',
        },
        log: {
          labelStartTime: 'Start',
          labelEndTime: 'End',
        },
    }
};

export const avatarInitialState: AvatarState[] = [
    {
      letter: 'N',
      color: 'orange',
    },
    {
      letter: 'A',
      color: 'purple',
    },
    {
      letter: 'T',
      color: 'orange',
    },
    {
      letter: 'J',
      color: 'purple',
    }
];