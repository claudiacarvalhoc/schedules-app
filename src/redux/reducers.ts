import { Reducer, combineReducers } from 'redux';
import { AppState } from './appState';
import { appReducer } from './app/reducer';

export const rootReducer = combineReducers({
    app: appReducer as Reducer<AppState>,
});

export type RootState = ReturnType<typeof rootReducer>;
export type DefaultRootState = ReturnType<typeof rootReducer>;