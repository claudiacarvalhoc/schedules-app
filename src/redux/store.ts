import { AnyAction } from 'redux';
import { AppState } from './appState';
import { ThunkAction, ThunkDispatch } from 'redux-thunk';
import { applyMiddleware, compose, createStore } from 'redux';
import { createLogger } from 'redux-logger';
import { rootReducer } from './reducers';
import thunk from 'redux-thunk';

export type AppThunk<R> = ThunkAction<R, AppState, unknown, AnyAction>;
export type AppDispatch = ThunkDispatch<AppState, unknown, AnyAction>;


export const configureStore = (initialState: AppState) => {
    const logger = createLogger();
    const composeEnhancers = (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
    const middleware = [thunk, logger];

    const store = createStore(
        rootReducer,
        {
            app: initialState,
        },
        composeEnhancers(applyMiddleware(...middleware)),
    );

    return store;
};

export default configureStore;
