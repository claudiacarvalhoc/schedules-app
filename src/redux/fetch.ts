import { AppDispatch } from './store';

export const getSchedules = (): ((dispatch: AppDispatch) => Promise<void>) => {
    return dispatch => {
        // TODO : ....
        return Promise.resolve();
    };
};

export const getScheduleLogs = (): ((dispatch: AppDispatch) => Promise<void>) => {
    return dispatch => {
        // TODO : ....
        return Promise.resolve();
    };
};
