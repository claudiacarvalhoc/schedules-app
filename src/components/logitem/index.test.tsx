import React from 'react';
import Enzyme, { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import toJson from 'enzyme-to-json';
import LogItem, { LogItemProps } from './index';
import { LogStatusState } from '../../redux/appState';
import thunk from 'redux-thunk';
import { appInitialState } from '../../redux/initialState';
import { Provider } from 'react-redux';

Enzyme.configure({ adapter: new Adapter() });

const getMountWrapper = (props: LogItemProps) => {
    const store = configureMockStore([thunk])({
        app: {
        ...appInitialState,
        scheduleLogs: [props.item],
        },
    });
    return mount(
        <Provider store={store}>
            <LogItem {...props} />
        </Provider>);
}

describe('<LogItem />', () => {
    it('render correctly', () => {
        const props = {
            item: {
                id: 1193520,
                startTime: new Date("2021-12-18T06:35:09.604Z"),
                endTime: new Date("2021-07-12T23:43:39.395Z"),
                status: LogStatusState.Running,
                serverName: "esse ad deserunt",
                scheduleId: 23802893,
            },
            texts: {
                labelStartTime: 'Start',
                labelEndTime: 'End',
            },
        };
        const component = getMountWrapper(props);
        expect(toJson(component)).toMatchSnapshot();
    });
});
