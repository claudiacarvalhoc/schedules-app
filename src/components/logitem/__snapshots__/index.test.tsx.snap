// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogItem /> render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(LogItem)
    item={
      Object {
        "endTime": 2021-07-12T23:43:39.395Z,
        "id": 1193520,
        "scheduleId": 23802893,
        "serverName": "esse ad deserunt",
        "startTime": 2021-12-18T06:35:09.604Z,
        "status": 4,
      }
    }
    texts={
      Object {
        "labelEndTime": "End",
        "labelStartTime": "Start",
      }
    }
  >
    <LogItem
      dispatch={[Function]}
      item={
        Object {
          "endTime": 2021-07-12T23:43:39.395Z,
          "id": 1193520,
          "scheduleId": 23802893,
          "serverName": "esse ad deserunt",
          "startTime": 2021-12-18T06:35:09.604Z,
          "status": 4,
        }
      }
      texts={
        Object {
          "labelEndTime": "End",
          "labelStartTime": "Start",
        }
      }
    >
      <WithStyles(ForwardRef(Card))
        className="logs"
        variant="outlined"
      >
        <ForwardRef(Card)
          className="logs"
          classes={
            Object {
              "root": "MuiCard-root",
            }
          }
          variant="outlined"
        >
          <WithStyles(ForwardRef(Paper))
            className="MuiCard-root logs"
            elevation={1}
            variant="outlined"
          >
            <ForwardRef(Paper)
              className="MuiCard-root logs"
              classes={
                Object {
                  "elevation0": "MuiPaper-elevation0",
                  "elevation1": "MuiPaper-elevation1",
                  "elevation10": "MuiPaper-elevation10",
                  "elevation11": "MuiPaper-elevation11",
                  "elevation12": "MuiPaper-elevation12",
                  "elevation13": "MuiPaper-elevation13",
                  "elevation14": "MuiPaper-elevation14",
                  "elevation15": "MuiPaper-elevation15",
                  "elevation16": "MuiPaper-elevation16",
                  "elevation17": "MuiPaper-elevation17",
                  "elevation18": "MuiPaper-elevation18",
                  "elevation19": "MuiPaper-elevation19",
                  "elevation2": "MuiPaper-elevation2",
                  "elevation20": "MuiPaper-elevation20",
                  "elevation21": "MuiPaper-elevation21",
                  "elevation22": "MuiPaper-elevation22",
                  "elevation23": "MuiPaper-elevation23",
                  "elevation24": "MuiPaper-elevation24",
                  "elevation3": "MuiPaper-elevation3",
                  "elevation4": "MuiPaper-elevation4",
                  "elevation5": "MuiPaper-elevation5",
                  "elevation6": "MuiPaper-elevation6",
                  "elevation7": "MuiPaper-elevation7",
                  "elevation8": "MuiPaper-elevation8",
                  "elevation9": "MuiPaper-elevation9",
                  "outlined": "MuiPaper-outlined",
                  "root": "MuiPaper-root",
                  "rounded": "MuiPaper-rounded",
                }
              }
              elevation={1}
              variant="outlined"
            >
              <div
                className="MuiPaper-root MuiCard-root logs MuiPaper-outlined MuiPaper-rounded"
              >
                <WithStyles(ForwardRef(CardContent))
                  className="content"
                >
                  <ForwardRef(CardContent)
                    className="content"
                    classes={
                      Object {
                        "root": "MuiCardContent-root",
                      }
                    }
                  >
                    <div
                      className="MuiCardContent-root content"
                    >
                      <div
                        className="header"
                      >
                        <WithStyles(ForwardRef(Typography))
                          className="title"
                          color="primary"
                          gutterBottom={true}
                          noWrap={true}
                        >
                          <ForwardRef(Typography)
                            className="title"
                            classes={
                              Object {
                                "alignCenter": "MuiTypography-alignCenter",
                                "alignJustify": "MuiTypography-alignJustify",
                                "alignLeft": "MuiTypography-alignLeft",
                                "alignRight": "MuiTypography-alignRight",
                                "body1": "MuiTypography-body1",
                                "body2": "MuiTypography-body2",
                                "button": "MuiTypography-button",
                                "caption": "MuiTypography-caption",
                                "colorError": "MuiTypography-colorError",
                                "colorInherit": "MuiTypography-colorInherit",
                                "colorPrimary": "MuiTypography-colorPrimary",
                                "colorSecondary": "MuiTypography-colorSecondary",
                                "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                "displayBlock": "MuiTypography-displayBlock",
                                "displayInline": "MuiTypography-displayInline",
                                "gutterBottom": "MuiTypography-gutterBottom",
                                "h1": "MuiTypography-h1",
                                "h2": "MuiTypography-h2",
                                "h3": "MuiTypography-h3",
                                "h4": "MuiTypography-h4",
                                "h5": "MuiTypography-h5",
                                "h6": "MuiTypography-h6",
                                "noWrap": "MuiTypography-noWrap",
                                "overline": "MuiTypography-overline",
                                "paragraph": "MuiTypography-paragraph",
                                "root": "MuiTypography-root",
                                "srOnly": "MuiTypography-srOnly",
                                "subtitle1": "MuiTypography-subtitle1",
                                "subtitle2": "MuiTypography-subtitle2",
                              }
                            }
                            color="primary"
                            gutterBottom={true}
                            noWrap={true}
                          >
                            <p
                              className="MuiTypography-root title MuiTypography-body1 MuiTypography-colorPrimary MuiTypography-noWrap MuiTypography-gutterBottom"
                            >
                              esse ad deserunt
                            </p>
                          </ForwardRef(Typography)>
                        </WithStyles(ForwardRef(Typography))>
                        <WithStyles(ForwardRef(Chip))
                          label={4}
                          size="small"
                        >
                          <ForwardRef(Chip)
                            classes={
                              Object {
                                "avatar": "MuiChip-avatar",
                                "avatarColorPrimary": "MuiChip-avatarColorPrimary",
                                "avatarColorSecondary": "MuiChip-avatarColorSecondary",
                                "avatarSmall": "MuiChip-avatarSmall",
                                "clickable": "MuiChip-clickable",
                                "clickableColorPrimary": "MuiChip-clickableColorPrimary",
                                "clickableColorSecondary": "MuiChip-clickableColorSecondary",
                                "colorPrimary": "MuiChip-colorPrimary",
                                "colorSecondary": "MuiChip-colorSecondary",
                                "deletable": "MuiChip-deletable",
                                "deletableColorPrimary": "MuiChip-deletableColorPrimary",
                                "deletableColorSecondary": "MuiChip-deletableColorSecondary",
                                "deleteIcon": "MuiChip-deleteIcon",
                                "deleteIconColorPrimary": "MuiChip-deleteIconColorPrimary",
                                "deleteIconColorSecondary": "MuiChip-deleteIconColorSecondary",
                                "deleteIconOutlinedColorPrimary": "MuiChip-deleteIconOutlinedColorPrimary",
                                "deleteIconOutlinedColorSecondary": "MuiChip-deleteIconOutlinedColorSecondary",
                                "deleteIconSmall": "MuiChip-deleteIconSmall",
                                "disabled": "Mui-disabled",
                                "icon": "MuiChip-icon",
                                "iconColorPrimary": "MuiChip-iconColorPrimary",
                                "iconColorSecondary": "MuiChip-iconColorSecondary",
                                "iconSmall": "MuiChip-iconSmall",
                                "label": "MuiChip-label",
                                "labelSmall": "MuiChip-labelSmall",
                                "outlined": "MuiChip-outlined",
                                "outlinedPrimary": "MuiChip-outlinedPrimary",
                                "outlinedSecondary": "MuiChip-outlinedSecondary",
                                "root": "MuiChip-root",
                                "sizeSmall": "MuiChip-sizeSmall",
                              }
                            }
                            label={4}
                            size="small"
                          >
                            <div
                              className="MuiChip-root MuiChip-sizeSmall"
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                            >
                              <span
                                className="MuiChip-label MuiChip-labelSmall"
                              >
                                4
                              </span>
                            </div>
                          </ForwardRef(Chip)>
                        </WithStyles(ForwardRef(Chip))>
                      </div>
                      <div
                        className="times"
                      >
                        <Time
                          label="Start"
                          time="December 18 2021"
                        >
                          <div>
                            <WithStyles(ForwardRef(Typography))
                              className="label"
                              gutterBottom={true}
                              variant="overline"
                            >
                              <ForwardRef(Typography)
                                className="label"
                                classes={
                                  Object {
                                    "alignCenter": "MuiTypography-alignCenter",
                                    "alignJustify": "MuiTypography-alignJustify",
                                    "alignLeft": "MuiTypography-alignLeft",
                                    "alignRight": "MuiTypography-alignRight",
                                    "body1": "MuiTypography-body1",
                                    "body2": "MuiTypography-body2",
                                    "button": "MuiTypography-button",
                                    "caption": "MuiTypography-caption",
                                    "colorError": "MuiTypography-colorError",
                                    "colorInherit": "MuiTypography-colorInherit",
                                    "colorPrimary": "MuiTypography-colorPrimary",
                                    "colorSecondary": "MuiTypography-colorSecondary",
                                    "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                    "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                    "displayBlock": "MuiTypography-displayBlock",
                                    "displayInline": "MuiTypography-displayInline",
                                    "gutterBottom": "MuiTypography-gutterBottom",
                                    "h1": "MuiTypography-h1",
                                    "h2": "MuiTypography-h2",
                                    "h3": "MuiTypography-h3",
                                    "h4": "MuiTypography-h4",
                                    "h5": "MuiTypography-h5",
                                    "h6": "MuiTypography-h6",
                                    "noWrap": "MuiTypography-noWrap",
                                    "overline": "MuiTypography-overline",
                                    "paragraph": "MuiTypography-paragraph",
                                    "root": "MuiTypography-root",
                                    "srOnly": "MuiTypography-srOnly",
                                    "subtitle1": "MuiTypography-subtitle1",
                                    "subtitle2": "MuiTypography-subtitle2",
                                  }
                                }
                                gutterBottom={true}
                                variant="overline"
                              >
                                <span
                                  className="MuiTypography-root label MuiTypography-overline MuiTypography-gutterBottom"
                                >
                                  Start
                                </span>
                              </ForwardRef(Typography)>
                            </WithStyles(ForwardRef(Typography))>
                            <div>
                              <WithStyles(ForwardRef(Chip))
                                label="December 18 2021"
                                size="small"
                              >
                                <ForwardRef(Chip)
                                  classes={
                                    Object {
                                      "avatar": "MuiChip-avatar",
                                      "avatarColorPrimary": "MuiChip-avatarColorPrimary",
                                      "avatarColorSecondary": "MuiChip-avatarColorSecondary",
                                      "avatarSmall": "MuiChip-avatarSmall",
                                      "clickable": "MuiChip-clickable",
                                      "clickableColorPrimary": "MuiChip-clickableColorPrimary",
                                      "clickableColorSecondary": "MuiChip-clickableColorSecondary",
                                      "colorPrimary": "MuiChip-colorPrimary",
                                      "colorSecondary": "MuiChip-colorSecondary",
                                      "deletable": "MuiChip-deletable",
                                      "deletableColorPrimary": "MuiChip-deletableColorPrimary",
                                      "deletableColorSecondary": "MuiChip-deletableColorSecondary",
                                      "deleteIcon": "MuiChip-deleteIcon",
                                      "deleteIconColorPrimary": "MuiChip-deleteIconColorPrimary",
                                      "deleteIconColorSecondary": "MuiChip-deleteIconColorSecondary",
                                      "deleteIconOutlinedColorPrimary": "MuiChip-deleteIconOutlinedColorPrimary",
                                      "deleteIconOutlinedColorSecondary": "MuiChip-deleteIconOutlinedColorSecondary",
                                      "deleteIconSmall": "MuiChip-deleteIconSmall",
                                      "disabled": "Mui-disabled",
                                      "icon": "MuiChip-icon",
                                      "iconColorPrimary": "MuiChip-iconColorPrimary",
                                      "iconColorSecondary": "MuiChip-iconColorSecondary",
                                      "iconSmall": "MuiChip-iconSmall",
                                      "label": "MuiChip-label",
                                      "labelSmall": "MuiChip-labelSmall",
                                      "outlined": "MuiChip-outlined",
                                      "outlinedPrimary": "MuiChip-outlinedPrimary",
                                      "outlinedSecondary": "MuiChip-outlinedSecondary",
                                      "root": "MuiChip-root",
                                      "sizeSmall": "MuiChip-sizeSmall",
                                    }
                                  }
                                  label="December 18 2021"
                                  size="small"
                                >
                                  <div
                                    className="MuiChip-root MuiChip-sizeSmall"
                                    onKeyDown={[Function]}
                                    onKeyUp={[Function]}
                                  >
                                    <span
                                      className="MuiChip-label MuiChip-labelSmall"
                                    >
                                      December 18 2021
                                    </span>
                                  </div>
                                </ForwardRef(Chip)>
                              </WithStyles(ForwardRef(Chip))>
                            </div>
                          </div>
                        </Time>
                        <Time
                          label="End"
                          time="July 13 2021"
                        >
                          <div>
                            <WithStyles(ForwardRef(Typography))
                              className="label"
                              gutterBottom={true}
                              variant="overline"
                            >
                              <ForwardRef(Typography)
                                className="label"
                                classes={
                                  Object {
                                    "alignCenter": "MuiTypography-alignCenter",
                                    "alignJustify": "MuiTypography-alignJustify",
                                    "alignLeft": "MuiTypography-alignLeft",
                                    "alignRight": "MuiTypography-alignRight",
                                    "body1": "MuiTypography-body1",
                                    "body2": "MuiTypography-body2",
                                    "button": "MuiTypography-button",
                                    "caption": "MuiTypography-caption",
                                    "colorError": "MuiTypography-colorError",
                                    "colorInherit": "MuiTypography-colorInherit",
                                    "colorPrimary": "MuiTypography-colorPrimary",
                                    "colorSecondary": "MuiTypography-colorSecondary",
                                    "colorTextPrimary": "MuiTypography-colorTextPrimary",
                                    "colorTextSecondary": "MuiTypography-colorTextSecondary",
                                    "displayBlock": "MuiTypography-displayBlock",
                                    "displayInline": "MuiTypography-displayInline",
                                    "gutterBottom": "MuiTypography-gutterBottom",
                                    "h1": "MuiTypography-h1",
                                    "h2": "MuiTypography-h2",
                                    "h3": "MuiTypography-h3",
                                    "h4": "MuiTypography-h4",
                                    "h5": "MuiTypography-h5",
                                    "h6": "MuiTypography-h6",
                                    "noWrap": "MuiTypography-noWrap",
                                    "overline": "MuiTypography-overline",
                                    "paragraph": "MuiTypography-paragraph",
                                    "root": "MuiTypography-root",
                                    "srOnly": "MuiTypography-srOnly",
                                    "subtitle1": "MuiTypography-subtitle1",
                                    "subtitle2": "MuiTypography-subtitle2",
                                  }
                                }
                                gutterBottom={true}
                                variant="overline"
                              >
                                <span
                                  className="MuiTypography-root label MuiTypography-overline MuiTypography-gutterBottom"
                                >
                                  End
                                </span>
                              </ForwardRef(Typography)>
                            </WithStyles(ForwardRef(Typography))>
                            <div>
                              <WithStyles(ForwardRef(Chip))
                                label="July 13 2021"
                                size="small"
                              >
                                <ForwardRef(Chip)
                                  classes={
                                    Object {
                                      "avatar": "MuiChip-avatar",
                                      "avatarColorPrimary": "MuiChip-avatarColorPrimary",
                                      "avatarColorSecondary": "MuiChip-avatarColorSecondary",
                                      "avatarSmall": "MuiChip-avatarSmall",
                                      "clickable": "MuiChip-clickable",
                                      "clickableColorPrimary": "MuiChip-clickableColorPrimary",
                                      "clickableColorSecondary": "MuiChip-clickableColorSecondary",
                                      "colorPrimary": "MuiChip-colorPrimary",
                                      "colorSecondary": "MuiChip-colorSecondary",
                                      "deletable": "MuiChip-deletable",
                                      "deletableColorPrimary": "MuiChip-deletableColorPrimary",
                                      "deletableColorSecondary": "MuiChip-deletableColorSecondary",
                                      "deleteIcon": "MuiChip-deleteIcon",
                                      "deleteIconColorPrimary": "MuiChip-deleteIconColorPrimary",
                                      "deleteIconColorSecondary": "MuiChip-deleteIconColorSecondary",
                                      "deleteIconOutlinedColorPrimary": "MuiChip-deleteIconOutlinedColorPrimary",
                                      "deleteIconOutlinedColorSecondary": "MuiChip-deleteIconOutlinedColorSecondary",
                                      "deleteIconSmall": "MuiChip-deleteIconSmall",
                                      "disabled": "Mui-disabled",
                                      "icon": "MuiChip-icon",
                                      "iconColorPrimary": "MuiChip-iconColorPrimary",
                                      "iconColorSecondary": "MuiChip-iconColorSecondary",
                                      "iconSmall": "MuiChip-iconSmall",
                                      "label": "MuiChip-label",
                                      "labelSmall": "MuiChip-labelSmall",
                                      "outlined": "MuiChip-outlined",
                                      "outlinedPrimary": "MuiChip-outlinedPrimary",
                                      "outlinedSecondary": "MuiChip-outlinedSecondary",
                                      "root": "MuiChip-root",
                                      "sizeSmall": "MuiChip-sizeSmall",
                                    }
                                  }
                                  label="July 13 2021"
                                  size="small"
                                >
                                  <div
                                    className="MuiChip-root MuiChip-sizeSmall"
                                    onKeyDown={[Function]}
                                    onKeyUp={[Function]}
                                  >
                                    <span
                                      className="MuiChip-label MuiChip-labelSmall"
                                    >
                                      July 13 2021
                                    </span>
                                  </div>
                                </ForwardRef(Chip)>
                              </WithStyles(ForwardRef(Chip))>
                            </div>
                          </div>
                        </Time>
                      </div>
                    </div>
                  </ForwardRef(CardContent)>
                </WithStyles(ForwardRef(CardContent))>
              </div>
            </ForwardRef(Paper)>
          </WithStyles(ForwardRef(Paper))>
        </ForwardRef(Card)>
      </WithStyles(ForwardRef(Card))>
    </LogItem>
  </Connect(LogItem)>
</Provider>
`;
